### PR TITLE
WIP 5150 flush content cache correctly

### LINKED ISSUE #3184: Dependabot can't resolve your PHP dependency files
Dependabot can't resolve your PHP dependency files.

As a result, Dependabot couldn't update your dependencies.

The error Dependabot encountered was:

```
Your requirements could not be resolved to an installable set of packages.
  Problem 1
    - neos/fusion-form v1.0.0 requires neos/fusion ^4.3 || ^5.0 || dev-master -> satisfiable by neos/fusion[4.3.0, 4.3.1, 4.3.10, 4.3.11, 4.3.12, 4.3.13, 4.3.14, 4.3.15, 4.3.16, 4.3.2, 4.3.3, 4.3.4, 4.3.5, 4.3.6, 4.3.7, 4.3.8, 4.3.9, 5.0.0, 5.0.1, 5.0.10, 5.0.11, 5.0.12, 5.0.13, 5.0.14, 5.0.2, 5.0.3, 5.0.4, 5.0.5, 5.0.6, 5.0.7, 5.0.8, 5.0.9, 5.1.0, 5.1.1, 5.1.10, 5.1.2, 5.1.3, 5.1.4, 5.1.5, 5.1.6, 5.1.7, 5.1.8, 5.1.9, 5.2.0, 5.2.1, 5.2.2, 5.2.3, 5.3.0].
    - neos/fusion-form v1.0.1 requires neos/fusion ^4.3 || ^5.0 || dev-master -> satisfiable by neos/fusion[4.3.0, 4.3.1, 4.3.10, 4.3.11, 4.3.12, 4.3.13, 4.3.14, 4.3.15, 4.3.16, 4.3.2, 4.3.3, 4.3.4, 4.3.5, 4.3.6, 4.3.7, 4.3.8, 4.3.9, 5.0.0, 5.0.1, 5.0.10, 5.0.11, 5.0.12, 5.0.13, 5.0.14, 5.0.2, 5.0.3, 5.0.4, 5.0.5, 5.0.6, 5.0.7, 5.0.8, 5.0.9, 5.1.0, 5.1.1, 5.1.10, 5.1.2, 5.1.3, 5.1.4, 5.1.5, 5.1.6, 5.1.7, 5.1.8, 5.1.9, 5.2.0, 5.2.1, 5.2.2, 5.2.3, 5.3.0].
    - neos/fusion-form v1.0.2 requires neos/fusion ^4.3 || ^5.0 || dev-master -> satisfiable by neos/fusion[4.3.0, 4.3.1, 4.3.10, 4.3.11, 4.3.12, 4.3.13, 4.3.14, 4.3.15, 4.3.16, 4.3.2, 4.3.3, 4.3.4, 4.3.5, 4.3.6, 4.3.7, 4.3.8, 4.3.9, 5.0.0, 5.0.1, 5.0.10, 5.0.11, 5.0.12, 5.0.13, 5.0.14, 5.0.2, 5.0.3, 5.0.4, 5.0.5, 5.0.6, 5.0.7, 5.0.8, 5.0.9, 5.1.0, 5.1.1, 5.1.10, 5.1.2, 5.1.3, 5.1.4, 5.1.5, 5.1.6, 5.1.7, 5.1.8, 5.1.9, 5.2.0, 5.2.1, 5.2.2, 5.2.3, 5.3.0].
    - neos/fusion-form v1.0.3 requires neos/fusion ^4.3 || ^5.0 || dev-master -> satisfiable by neos/fusion[4.3.0, 4.3.1, 4.3.10, 4.3.11, 4.3.12, 4.3.13, 4.3.14, 4.3.15, 4.3.16, 4.3.2, 4.3.3, 4.3.4, 4.3.5, 4.3.6, 4.3.7, 4.3.8, 4.3.9, 5.0.0, 5.0.1, 5.0.10, 5.0.11, 5.0.12, 5.0.13, 5.0.14, 5.0.2, 5.0.3, 5.0.4, 5.0.5, 5.0.6, 5.0.7, 5.0.8, 5.0.9, 5.1.0, 5.1.1, 5.1.10, 5.1.2, 5.1.3, 5.1.4, 5.1.5, 5.1.6, 5.1.7, 5.1.8, 5.1.9, 5.2.0, 5.2.1, 5.2.2, 5.2.3, 5.3.0].
    - neos/fusion-form v1.0.4 requires neos/fusion ^4.3 || ^5.0 || dev-master -> satisfiable by neos/fusion[4.3.0, 4.3.1, 4.3.10, 4.3.11, 4.3.12, 4.3.13, 4.3.14, 4.3.15, 4.3.16, 4.3.2, 4.3.3, 4.3.4, 4.3.5, 4.3.6, 4.3.7, 4.3.8, 4.3.9, 5.0.0, 5.0.1, 5.0.10, 5.0.11, 5.0.12, 5.0.13, 5.0.14, 5.0.2, 5.0.3, 5.0.4, 5.0.5, 5.0.6, 5.0.7, 5.0.8, 5.0.9, 5.1.0, 5.1.1, 5.1.10, 5.1.2, 5.1.3, 5.1.4, 5.1.5, 5.1.6, 5.1.7, 5.1.8, 5.1.9, 5.2.0, 5.2.1, 5.2.2, 5.2.3, 5.3.0].
    - don't install neos/fusion 5.0.0|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.1|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.10|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.11|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.12|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.13|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.14|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.2|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.3|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.4|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.5|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.6|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.7|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.8|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.0.9|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.0|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.1|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.10|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.2|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.3|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.4|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.5|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.6|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.7|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.8|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.1.9|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.2.0|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.2.1|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.2.2|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.2.3|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 5.3.0|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.0|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.1|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.10|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.11|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.12|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.13|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.14|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.15|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.16|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.2|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.3|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.4|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.5|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.6|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.7|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.8|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - don't install neos/fusion 4.3.9|remove neos/neos-development-collection No version set (parsed as 1.0.0)
    - Installation request for neos/neos-development-collection No version set (parsed as 1.0.0) -> satisfiable by neos/neos-development-collection[No version set (parsed as 1.0.0)].
    - Installation request for neos/fusion-form ^1.0 -> satisfiable by neos/fusion-form[v1.0.0, v1.0.1, v1.0.2, v1.0.3, v1.0.4].

```

If you think the above is an error on Dependabot's side please don't hesitate to get in touch - we'll do whatever we can to fix it.

[View the update logs](https://app.dependabot.com/accounts/neos/update-logs/55019333).

### LINKED ISSUE #5150: BUG: Discarded node remains in cache if UI was reloaded between creation and discard
I've encountered some weird occasions in which the content cache becomes stale after discarding.

For example:

If I create a content node and then immediately discard it, the node remains cached:

https://github.com/neos/neos-ui/assets/2522299/edb77745-e3d5-4fb7-86a2-0be4145485bd

If I create a content node, then *edit its properties* and immediately discard it, the node vanishes from cache (which is what should always happen):

https://github.com/neos/neos-ui/assets/2522299/309b0458-80ac-498e-a8cd-a21a7232c427

Now, weirdly, if I create a content node, then edit its properties, then *reload the UI* and discard it, the node remains cached:

https://github.com/neos/neos-ui/assets/2522299/2d6bb328-fa94-4915-93c3-b777a1fb8fb9

### PR DESCRIPTION
**Pull request by mhsdesign**
Experimental possible alternative to https://github.com/neos/neos-development-collection/pull/5155

Resolves: https://github.com/neos/neos-development-collection/issues/5150

The discard tests of the neos ui will pass (#5150)

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php [hunk 1]
```diff
@@ -133,19 +135,28 @@ public function onBeforeEvent(EventInterface $eventInstance, EventEnvelope $even
...
+        if ($eventInstance instanceof WorkspaceWasPartiallyDiscarded) {
+            foreach ($eventInstance->discardedNodes as $discardedNode) {
+                $this->scheduleCacheFlushJobForNodeAggregate(
+                    $this->contentRepository,
+                    $eventInstance->workspaceName,
+                    $discardedNode->nodeAggregateId
+                );
+            }
+        }
+
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php:147]: fixes one ui test

```
FIX #3184: Discarded node move changes are reflected correctly in the document tree
Scenario #2: Moved nodes do not just disappear after discarding the move change
```

File: Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php [hunk 2]
```diff
@@ -133,19 +135,28 @@ public function onBeforeEvent(EventInterface $eventInstance, EventEnvelope $even
...
             $eventInstance instanceof NodeAggregateWasRemoved
             // NOTE: when moving a node, we need to clear the cache not just after the move was completed,
             // but also on the original location. Otherwise, we have the problem that the cache is not
             // cleared, leading to presumably duplicate nodes in the UI.
             || $eventInstance instanceof NodeAggregateWasMoved
         ) {
-            $workspace = $this->contentRepository->getWorkspaceFinder()->findOneByCurrentContentStreamId($eventInstance->getContentStreamId());
-            if ($workspace === null) {
+            try {
+                $contentGraph = $this->contentRepository->getContentGraph($eventInstance->workspaceName);
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Fusion/Cache/GraphProjectorCatchUpHookForCacheFlushing.php:156]: using `$eventInstance->workspaceName` fixes most of the ui discard failures

**Comment by mhsdesign** [comment]
So the ui is green with this but now the new `NodesInUserWorkspace.feature` tests (thanks Wilhelm) dont pass at all ... neither for WorkspaceWasPartiallyDiscarded nor on WorkspaceWasDiscarded  ...

**Comment by dlubitz** [comment]
As this is fixed now properly, I close this one.

**Pull request closed without merging**